### DISCUSSION TITLE
Ultralytics YOLO only uses a subset of the images for training

### DISCUSSION THREAD

**Discussion by breedaditi** [ultralytics/ultralytics#13577] [Q&A]
Hello,
            I am trying to understand why the YOLOv5 model in ultralytics is only using a subset of the images for training.

When i run the code , it shows the following:

train: Scanning FileLocation\labels\train... 2614 
train: New cache created: FileLocation\labels\train.cache

So it confirms I have 2614 images in the train folder for which it created a cache. 

However when it starts training, it shows:

![image](https://github.com/ultralytics/ultralytics/assets/102319676/815ab6b7-6449-4175-91bb-e59e30324759)

The images and instance number does not change for all epochs.

The machine does not have memory limitation. This machine has a 24GB GPU. 

Can someone please help me understand ?

Thanks

**Comment by glenn-jocher** [comment]
@breedaditi hello,

Thank you for reaching out and providing detailed information about your issue. Let's work through this together.

Firstly, it's important to ensure that you are using the latest versions of both `torch` and `ultralytics`. If you haven't already, please upgrade your packages with the following commands:

```bash
pip install --upgrade torch ultralytics
```

Next, it would be helpful to have a minimum reproducible code example to better understand the context and reproduce the issue on our end. This will allow us to investigate and provide a more accurate solution. You can find guidelines for creating a minimum reproducible example here: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example).

From your description, it seems like the training process is not utilizing all the images in your dataset. Here are a few things to check:

1. **Dataset Configuration**: Ensure that your dataset configuration file (e.g., `data.yaml`) correctly points to the directories containing your images and labels. The paths should be relative to the location of the YAML file.

2. **Caching**: The cache creation message indicates that the labels were successfully scanned. However, during training, if the number of images and instances remains constant across epochs, it might suggest that the dataset is not being fully utilized. Verify that there are no issues with the dataset loading process.

3. **Batch Size and DataLoader**: Check your batch size and DataLoader settings. If the batch size is too large, it might not iterate over all images. You can adjust the batch size in your training command:

    ```bash
    yolo train data=your_data.yaml model=yolov5s.pt batch=16 epochs=100 imgsz=640
    ```

4. **Data Augmentation**: Ensure that data augmentation settings are not inadvertently limiting the dataset. Sometimes, aggressive augmentation can lead to fewer unique images being used per epoch.

If you continue to experience issues, please share the relevant parts of your dataset configuration and training script. This will help us diagnose the problem more effectively.

Looking forward to your response!
